python/lsst/ts/envsensors/device/rpi_serial_hat.py

```diff
@@ -63,34 +112,138 @@ def __init__(
             callback_func=callback_func,
             log=log,
         )
+        if self._device_id in RpiSerialHat.serial_ports:
+            (
+                self._ser_port,
+                self._pin_on,
+                self._pin_din,
+                self._pin_dout,
+                self._pin_dirn,
+            ) = RpiSerialHat.serial_ports[self._device_id]
```

Comment:
This is scary. The `serial_ports` function is, in my opinion, returning too many values. I strongly suggest you return an instance of a type, such as SerialPortInfo. Then save that as a named instance and look up the values within the instance. E.g.:
```
self._serial_info = RpiSerialHat.serial_port_info[self._device_id]
# then access values using, e.g.: self._serial_info.port
```

Reply:
Here `serial_ports` is a dict, not a function. But I agree that the code is not very transparent. I prefer to leave it as is for now and would like to discuss a better implementation once you're back from vacation.